### ISSUE TITLE
Support advertising a different address than the one bound to for network configurations

### ISSUE THREAD

**Issue by npepinpe** [zeebe-io/zeebe#3143]
**Is your feature request related to a problem? Please describe.**

It is currently difficult to set up Zeebe brokers behind a proxy (e.g. a service mesh) as the address we bind the various ports to is the one we advertise to the other brokers; this means using [linkerd](https://linkerd.io) or anything else that transparently proxies traffic to and from the brokers will not work out of the box for Zeebe.

As service mesh are increasingly popular (with good reasons) in the k8s ecosystem, it makes sense for us to support them.

**Describe the solution you'd like**

An optional configuration option `advertisedHost` for every `NetworkCfg` which allows us to override the advertised host (i.e. what other nodes should use as endpoint) as opposed to the "host" param which is the address we will be ourselves to, as well as a default one (which can be overriden using `ZEEBE_ADVERTISED_HOST`).

**Describe alternatives you've considered**

We already have an alternative solution, which is to use port offset/different ports, but it is solidly in the realm of workarounds.

**Comment by IcyEagle** [comment]
@npepinpe Hi there!

It was included in 0.21.1, right?

Unfortunately, it doesn't work for me. I use the next startup script:
```
startup-sh: |
    #!/bin/bash -xeu

    configFile=/usr/local/zeebe/conf/zeebe.cfg.toml
    export ZEEBE_ADVERTISED_HOST=$(hostname -f)
    export ZEEBE_HOST="0.0.0.0"
    export ZEEBE_NODE_ID="${HOSTNAME##*-}"

    # We need to specify all brokers as contact points for partition healing to work correctly
    # https://github.com/zeebe-io/zeebe/issues/2684
    ZEEBE_CONTACT_POINTS=${HOSTNAME::-1}0.$(hostname -d):26502
    for (( i=1; i<$ZEEBE_CLUSTER_SIZE; i++ ))
    do
        ZEEBE_CONTACT_POINTS="${ZEEBE_CONTACT_POINTS},${HOSTNAME::-1}$i.$(hostname -d):26502"
    done
    export ZEEBE_CONTACT_POINTS="${ZEEBE_CONTACT_POINTS}"

    exec /usr/local/zeebe/bin/broker
```

Then I check cluster status via _zbctl_:
```
Cluster size: 3
Partitions count: 3
Replication factor: 3
Brokers:
  Broker 0 - 0.0.0.0:26501
  Broker 1 - 0.0.0.0:26501
  Broker 2 - 0.0.0.0:26501
```

my config file:
```
zeebe.cfg.toml: |
    # For more information about this configuration visit:
    [threads]
    cpuThreadCount = 1

    [gateway.monitoring]
    enabled = true

    [[exporters]]
    id = "hazelcast"
    className = "org.project.HazelcastExporter"

    [exporters.args]
    host = "core.dev.svc.cluster.local"
    enabledValueTypes = "JOB,WORKFLOW_INSTANCE,DEPLOYMENT,INCIDENT,TIMER,VARIABLE,MESSAGE,MESSAGE_SUBSCRIPTION,MESSAGE_START_EVENT_SUBSCRIPTION"
```

Where am I wrong?

**Comment by npepinpe** [comment]
Can you open an issue and share the environment (just the ZEEBE_* vars minus any sensitive info)? Also check if there are any exceptions popping up in your logs on the brokers. As a side note, if you're setting the host to a static value (e.g. `0.0.0.0`), it's probably simpler to add that to the configuration file under `[network]`.
